src/UI/Art/SpriteArtist/SpriteLibrary.java

```diff
@@ -166,90 +151,90 @@ public static TerrainSpriteSet getTerrainSpriteSet(Location loc)
    */
   private static void createTerrainSpriteSet(SpriteSetKey spriteKey)
   {
-    BaseTerrain terrainType = spriteKey.terrainKey;
+    TerrainType terrainType = spriteKey.terrainKey;
     System.out.println("INFO: Loading terrain sprites for " + terrainType);
 
     TerrainSpriteSet ss = null;
     int w = baseSpriteSize;
     int h = baseSpriteSize;
 
-    // This is for sure kinda ugly, but not much more than the equivalent switch/case
-    // This *could* be moved into BaseTerrain, but I feel like perhaps that would be too close integration of model and view (though it'd make it nice and simple in the short term?)
-    if( terrainType == Bridge.getInstance() )
+    // Load the appropriate images and define the necessary relationships for each tile type.
```

Comment:
Perhaps having integer indices wouldn't be a bad idea...
Switch/case is IMO a bit more readable than if/else, and you seem determined to keep UI-specific knowledge (sprites and such especially) out of TerrainType.
Thoughts?

Reply:
Hah, this after you were so dead-set on removing the switch. At this point, I don't think I want to go back to enums, and adding integer indices just so we can do a switch feels backwards to me. I agree it isn't quite as nice and compact as the switch statement was, but it's not too horrible.

Reply:
I'll live, I guess.